syntax = "proto3";

package node;

// Interface exported by the server.
service Node {
  // A simple RPC.
  //
  // Restarts all services and generates new keys for a new round
  rpc NewRound(MsgReq) returns (MsgReq) {}

  // A server-to-client streaming RPC.
  // server sends messages and the client request them
  // Obtains the Messages available.  Results are
  // streamed rather than returned at once
  rpc DumpMessages(MsgReq) returns (stream Msg) {}

  // A client-to-server streaming RPC.
  //
  // Send messages to the server
  rpc PutMessages(stream Msg) returns (MsgReq) {}
}

message Msg {
    bytes data = 1;
}

message MsgReq{
    bool yes = 1;
}

